src/storage/table.cc

```diff
@@ -114,55 +115,33 @@ void DBTable::InsertRecords(
   int data_size = 0;
   auto block = GetBlockForInsert();
   int offset = 0;
-  int length = l;
-  // optimize?
-  int column_types[num_cols];
-  for (int i = 0; i < num_cols; i++) {
-    column_types[i] = schema->field(i)->type()->id();
-  }
+
+  // TODO: (Optimize) Why don't we switch column type then loop over rows?
```

Comment:
@srsuryadev Why don't we switch column type then loop over rows? If the lambda function is not inlined, we may pay a huge cost here.

Reply:
I don't see any reason, it can be modified to our needs!

Reply:
I don't see any reason, it can be modified to our needs!

Reply:
We can open another issue to address this. I don't have a good idea to break this in this PR.